src/Engine/IO/Window.cpp

```diff
@@ -117,14 +150,28 @@ void Window::update(float dt) {
     Mouse::update(x, y);
 
     /* Update ImGui */
+    // TODO: clean up this code
+    static bool imGuiActive = false;
+    static bool priorCursorState;
     imGuiTimer += dt;
-    if (Keyboard::isKeyPressed(GLFW_KEY_GRAVE_ACCENT) && 
-       (Keyboard::isKeyPressed(GLFW_KEY_LEFT_SHIFT) || Keyboard::isKeyPressed(GLFW_KEY_RIGHT_SHIFT)) &&
-        imGuiTimer >= 0.5) {
+    if (Keyboard::isKeyPressed(GLFW_KEY_GRAVE_ACCENT) && imGuiTimer >= 0.5) {
+        if (!imGuiActive) {
+            priorCursorState = cursorEnabled;
+            setCursorEnabled(true);
+        }
+
         toggleImGui();
         imGuiTimer = 0.0;
+        imGuiActive = !imGuiActive;
+        ImGui_ImplGlfwGL3_NewFrame(isImGuiEnabled());
+
+        if (!imGuiActive) {
+            setCursorEnabled(priorCursorState);
+        }
+    }
+    else {
+        ImGui_ImplGlfwGL3_NewFrame(isImGuiEnabled());
```

Comment:
I think this should be executed per frame, not only if the the ~ key is pressed

Reply:
it is. that's what the else is there for